src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java

```diff
@@ -0,0 +1,360 @@
+                {
+                    return server.getPlayerList().getPlayerByUsername(potentialArgumentValue);
+                }
+                else
+                {
+                    if ("[abandoned]".equals(potentialArgumentValue))
+                    {
+                        return new FakePlayer(server.getWorld(0), new GameProfile(UUID.randomUUID(), "[abandoned]"));
+                    }
+                    return null;
+                }
+            case COLONY:
+                final List<String> colonyNumberStrings = getColonyIdStrings();
+                try
+                {
+                    int colonyNumber = Integer.parseInt(potentialArgumentValue);
+                    if (sender instanceof EntityPlayer)
+                    {
+                        if (colonyNumber == -1)
+                        {
+                            final IColony icolony = ColonyManager.getIColonyByOwner(sender.getEntityWorld(), (EntityPlayer) sender);
+                            if (icolony != null)
+                            {
+                                colonyNumber = icolony.getID();
+                            }
+                        }
+                    }
+                    if (colonyNumberStrings.contains(String.valueOf(colonyNumber)))
+                    {
+                        return ColonyManager.getColony(colonyNumber);
+                    }
+                }
+                catch (final NumberFormatException e)
+                {
+                    return null;
+                }
+            case CITIZEN:
+                final ArrayList<ActionMenuHolder> reversedParsedHolderList = new ArrayList<>(parsedHolders);
+                Collections.reverse(reversedParsedHolderList);
+                Colony colony = null;
+                for (final ActionMenuHolder actionMenuHolder : reversedParsedHolderList)
+                {
+                    final ActionArgument actionArgument = actionMenuHolder.getActionArgument();
+                    if (actionArgument.getType() == COLONY)
+                    {
+                        colony = (Colony) actionMenuHolder.getValue();
+                        break;
+                    }
+                }
+                if (null != colony)
+                {
+                    final ICitizenManager citizenManager = colony.getCitizenManager();
+                    return findCitizenForCitizenManager(citizenManager, potentialArgumentValue);
+                }
+                else
+                {
+                    // no colony specified for citizen
+                    final List<Colony> colonyList = ColonyManager.getColonies();
+                    for (final Colony someColony : colonyList)
+                    {
+                        final ICitizenManager citizenManager = someColony.getCitizenManager();
+                        return findCitizenForCitizenManager(citizenManager, potentialArgumentValue);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this "return" statement or make it conditional. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1751)